protocol/app/ante.go

```diff
@@ -233,8 +228,8 @@ func (h *lockingAnteHandler) clobAnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 	// We now acquire the global ante handler since the clob decorator is not thread safe and performs
 	// several reads and writes across many stores.
 	if !simulate && (ctx.IsCheckTx() || ctx.IsReCheckTx()) {
-		h.globalLock.Lock()
-		defer h.globalLock.Unlock()
+		ctx.MultiStore().(cachemulti.Store).Lock(locking.GlobalKey)
```

Comment:
There's no locking semantics changes in this PR right? 

Just to confirm: previously, it's not thread safe for `clob.AnteHandle` to directly modify `AuthStore` and this is still true now

Reply:
Correct. Taking the global lock here means taking a lock for everything except AuthStore, which is managed by the previous level of locks.